src/plus/integrations/providers/github.ts

```diff
@@ -317,3 +318,53 @@ export class GitHubEnterpriseIntegration extends GitHubIntegrationBase<SelfHoste
 		return super.connect(source);
 	}
 }
+
+export type GitHubPullRequestURLIdentity = {
+	ownerAndRepo?: string;
+	prNumber?: string;
+};
+
+export function getPullRequestIdentityValuesFromSearch(search: string): GitHubPullRequestURLIdentity {
+	let ownerAndRepo: string | undefined = undefined;
+	let prNumber: string | undefined = undefined;
+
+	let match = search.match(/([^/]+\/[^/]+)\/pull\/(\d+)/); // with org and rep name
+	if (match != null) {
+		ownerAndRepo = match[1];
+		prNumber = match[2];
+	}
+
+	if (prNumber == null) {
+		match = search.match(/(?:\/|^)pull\/(\d+)/); // without repo name
+		if (match != null) {
+			prNumber = match[1];
+		}
+	}
+
+	if (prNumber == null) {
+		match = search.match(/(?:#|\/)(\d+)/); // any number starting with # or /
+		if (match != null) {
+			prNumber = match[1];
+		}
+	}
+
+	return { ownerAndRepo: ownerAndRepo, prNumber: prNumber };
+}
```

Comment:
Same as with the function below, I suggest moving these to `pullRequest.ts` and, as soon as we start considering other providers, making them more generic/flexible.

Perhaps then the GitHub-specific regexes we are using can stay in GitHub-domain files like this one, and be imported there.

Reply:
@axosoft-ramint 

fixed